docs: improve ApplicationRef.bootstrap docs

umairhm opened the pull request:
add proper examples and update usage notes
add references to the method at relevant places in the docs

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: [41791](https://github.com/angular/angular/issues/41791)


## What is the new behavior?
Improved the documentation of [ApplicationRef.bootstrap](https://angular.io/api/core/ApplicationRef#bootstrap) method.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

kapunahelewong committed:
docs: edit copy for tick() in testing and api docs (#35697)

Fixes #35696

PR Close #35697

AleksanderBodurri committed:
feat(docs-infra): port over material io theming logic to angular io (#41129)

brings in theming tools from material io into angular io in preparation of implementing darkmode

PR Close #41129

AleksanderBodurri committed:
refactor(docs-infra): rename namespaced scss to index files (#41129)

scss files were forwarded from files that were named without convention, changes these file names to follow conventions

PR Close #41129

AleksanderBodurri committed:
refactor(docs-infra): grab scss constants using scss use syntax instead of the global imports (#41129)

move away from using global constants in scss files because @import will be deprecated soon

PR Close #41129

AleksanderBodurri committed:
refactor(docs-infra): grab scss mixins with scss use syntax instead of global imports (#41129)

move away from global mixins because @import is going to be deprecated

PR Close #41129

AleksanderBodurri committed:
feat(docs-infra): create typography and layout theme files (#41129)

creates theming files for the aio typography styles; done as part of the effort to make aio themeable

PR Close #41129

AleksanderBodurri committed:
feat(docs-infra): create module theme files (#41129)

creates theming files for the module styles in aio; done as part of the effor to make aio themeable

PR Close #41129

AleksanderBodurri committed:
feat(docs-infra): implement dark mode (#41129)

defines styles for a first iteration of an aio darkmode

PR Close #41129

iRealNirmal committed:
fix(platform-browser): update started state on reset (#41608)

This commit fixes the state of variable _started on call of reset method.

Earlier behaviour was on call of `reset()` method we are not setting back
`_started` flag's value to false and it created various issue for end user
which were expecting this behaviour as per name of method.

We provided end user `reset()` method, but it was empty method and on call
of it wasn't doing anything. As end user/developer were not able to
reuse animation not animation after call of reset method.

In this PR on call of `reset()` method we are setting flag `_started` value to false
which can be accessed by `hasStarted()`.

Resolves #18140

PR Close #41608

TeriGlover committed:
docs: Edited to remove jargon. (#41978)

PR Close #41978

TeriGlover committed:
docs: edits to remove jargon (#42025)

PR Close #42025

TeriGlover committed:
docs: Edits to remove jargon (#42026)

PR Close #42026

TeriGlover committed:
docs: Edits to remove jargon (#42029)

PR Close #42029

TeriGlover committed:
docs: Edits to remove jargon in Reference section (#42033)

PR Close #42033

lacolaco committed:
build(docs-infra): allow only Node.js v14 and above (#42093)

The AIO doc-gen uses JavaScript syntax that requires node.js 14.0.0 or above.

Closes #42086

PR Close #42093

TeriGlover committed:
docs: Edited to remove jargon, fixed issue (#42026)

PR Close #42026

josephperrott committed:
build(animations): update supported range of node versions to be less restrictive (#42205)

Update the supported range of node versions for to be less restrictive, no longer causing
yarn or npm to fail engine's checks for future versions of node.

While this change will no longer cause yarn or npm to fail these engine's check, this does
not reflect a change in the officially supported versions of node for Angular.  Angular
continues to maintain support for Active LTS and Maintenance LTS versions of node.

PR Close #42205

josephperrott committed:
build(bazel): update supported range of node versions to be less restrictive (#42205)

Update the supported range of node versions for to be less restrictive, no longer causing
yarn or npm to fail engine's checks for future versions of node.

While this change will no longer cause yarn or npm to fail these engine's check, this does
not reflect a change in the officially supported versions of node for Angular.  Angular
continues to maintain support for Active LTS and Maintenance LTS versions of node.

PR Close #42205

josephperrott committed:
build(common): update supported range of node versions to be less restrictive (#42205)

Update the supported range of node versions for to be less restrictive, no longer causing
yarn or npm to fail engine's checks for future versions of node.

While this change will no longer cause yarn or npm to fail these engine's check, this does
not reflect a change in the officially supported versions of node for Angular.  Angular
continues to maintain support for Active LTS and Maintenance LTS versions of node.

PR Close #42205

josephperrott committed:
build(compiler-cli): update supported range of node versions to be less restrictive (#42205)

Update the supported range of node versions for to be less restrictive, no longer causing
yarn or npm to fail engine's checks for future versions of node.

While this change will no longer cause yarn or npm to fail these engine's check, this does
not reflect a change in the officially supported versions of node for Angular.  Angular
continues to maintain support for Active LTS and Maintenance LTS versions of node.

PR Close #42205

josephperrott committed:
build(compiler): update supported range of node versions to be less restrictive (#42205)

Update the supported range of node versions for to be less restrictive, no longer causing
yarn or npm to fail engine's checks for future versions of node.

While this change will no longer cause yarn or npm to fail these engine's check, this does
not reflect a change in the officially supported versions of node for Angular.  Angular
continues to maintain support for Active LTS and Maintenance LTS versions of node.

PR Close #42205

josephperrott committed:
build(elements): update supported range of node versions to be less restrictive (#42205)

Update the supported range of node versions for to be less restrictive, no longer causing
yarn or npm to fail engine's checks for future versions of node.

While this change will no longer cause yarn or npm to fail these engine's check, this does
not reflect a change in the officially supported versions of node for Angular.  Angular
continues to maintain support for Active LTS and Maintenance LTS versions of node.

PR Close #42205

josephperrott committed:
build(forms): update supported range of node versions to be less restrictive (#42205)

Update the supported range of node versions for to be less restrictive, no longer causing
yarn or npm to fail engine's checks for future versions of node.

While this change will no longer cause yarn or npm to fail these engine's check, this does
not reflect a change in the officially supported versions of node for Angular.  Angular
continues to maintain support for Active LTS and Maintenance LTS versions of node.

PR Close #42205

josephperrott committed:
build(language-service): update supported range of node versions to be less restrictive (#42205)

Update the supported range of node versions for to be less restrictive, no longer causing
yarn or npm to fail engine's checks for future versions of node.

While this change will no longer cause yarn or npm to fail these engine's check, this does
not reflect a change in the officially supported versions of node for Angular.  Angular
continues to maintain support for Active LTS and Maintenance LTS versions of node.

PR Close #42205

josephperrott committed:
build(localize): update supported range of node versions to be less restrictive (#42205)

Update the supported range of node versions for to be less restrictive, no longer causing
yarn or npm to fail engine's checks for future versions of node.

While this change will no longer cause yarn or npm to fail these engine's check, this does
not reflect a change in the officially supported versions of node for Angular.  Angular
continues to maintain support for Active LTS and Maintenance LTS versions of node.

PR Close #42205

josephperrott committed:
build(platform-browser-dynamic): update supported range of node versions to be less restrictive (#42205)

Update the supported range of node versions for to be less restrictive, no longer causing
yarn or npm to fail engine's checks for future versions of node.

While this change will no longer cause yarn or npm to fail these engine's check, this does
not reflect a change in the officially supported versions of node for Angular.  Angular
continues to maintain support for Active LTS and Maintenance LTS versions of node.

PR Close #42205

josephperrott committed:
build(platform-browser): update supported range of node versions to be less restrictive (#42205)

Update the supported range of node versions for to be less restrictive, no longer causing
yarn or npm to fail engine's checks for future versions of node.

While this change will no longer cause yarn or npm to fail these engine's check, this does
not reflect a change in the officially supported versions of node for Angular.  Angular
continues to maintain support for Active LTS and Maintenance LTS versions of node.

PR Close #42205

josephperrott committed:
build(platform-server): update supported range of node versions to be less restrictive (#42205)

Update the supported range of node versions for to be less restrictive, no longer causing
yarn or npm to fail engine's checks for future versions of node.

While this change will no longer cause yarn or npm to fail these engine's check, this does
not reflect a change in the officially supported versions of node for Angular.  Angular
continues to maintain support for Active LTS and Maintenance LTS versions of node.

PR Close #42205

josephperrott committed:
build(router): update supported range of node versions to be less restrictive (#42205)

Update the supported range of node versions for to be less restrictive, no longer causing
yarn or npm to fail engine's checks for future versions of node.

While this change will no longer cause yarn or npm to fail these engine's check, this does
not reflect a change in the officially supported versions of node for Angular.  Angular
continues to maintain support for Active LTS and Maintenance LTS versions of node.

PR Close #42205

josephperrott committed:
build(service-worker): update supported range of node versions to be less restrictive (#42205)

Update the supported range of node versions for to be less restrictive, no longer causing
yarn or npm to fail engine's checks for future versions of node.

While this change will no longer cause yarn or npm to fail these engine's check, this does
not reflect a change in the officially supported versions of node for Angular.  Angular
continues to maintain support for Active LTS and Maintenance LTS versions of node.

PR Close #42205

josephperrott committed:
build(upgrade): update supported range of node versions to be less restrictive (#42205)

Update the supported range of node versions for to be less restrictive, no longer causing
yarn or npm to fail engine's checks for future versions of node.

While this change will no longer cause yarn or npm to fail these engine's check, this does
not reflect a change in the officially supported versions of node for Angular.  Angular
continues to maintain support for Active LTS and Maintenance LTS versions of node.

PR Close #42205

josephperrott committed:
build(core): update supported range of node versions to be less restrictive (#42205)

Update the supported range of node versions for to be less restrictive, no longer causing
yarn or npm to fail engine's checks for future versions of node.

While this change will no longer cause yarn or npm to fail these engine's check, this does
not reflect a change in the officially supported versions of node for Angular.  Angular
continues to maintain support for Active LTS and Maintenance LTS versions of node.

PR Close #42205

iRealNirmal committed:
fix(forms): Add float number support for min and max validator (#42223)

Added float number support for minValidator and maxValidator

Closes #42215

PR Close #42223

josephperrott committed:
refactor(dev-infra): move the release notes into its own directory under release (#42225)

Move the release notes tooling under its own directory under release rather than
within publish, in preparation to have a release note generation command for ad-hoc
release note generation.

PR Close #42225

josephperrott committed:
feat(dev-infra): add release notes generation to ng-dev (#42225)

Adds tooling to create ad-hoc release note entries via `ng-dev release notes`.

PR Close #42225

josephperrott committed:
docs: remove/update broken links in resources appearing on aio (#42232)

Remove or update broken links to resources from the resources and
contributors pages on aio.

Closes #39719

PR Close #42232

devversion committed:
fix(dev-infra): format command incorrectly prints outdated files (#42252)

When `ng-dev format --check` is run, the ng-dev tool prints out
all files that are out-of-date. We recently updated the format
tool to also capture `stderr` for failed files. This broke the
console message as we did not unwrap the `FormatFailure` to
their file path when printing the "ng-dev format" fix command.

PR Close #42252

petebacondarwin committed:
docs: remove VS 2015 guide (#42253)

This guide is outdated and doesn't appear in the navigation.

Fixes #17320

PR Close #42253

petebacondarwin committed:
docs: convert the last `my-app` components to `app-root` (#42256)

Most of these were fixed in other PRs but there were are couple of stragglers.
Note that `my-app` components in non-documentation facing code, such as
compliance tests have not been changed.

Fixes #20235

PR Close #42256

crisbeto committed:
test: clean up internal testing utilities (#42177)

We have some internal proxies for all of the Jasmine functions, as well as some other helpers. This code hasn't been touched in more than 5 years, it can lead to confusion and it isn't really necessary since the same can be achieved using Jasmine.

These changes remove most of the code and clean up our existing unit tests.

PR Close #42177

atscott committed:
docs: clarify location.subscribe behavior (#42272)

`Location.go` does not trigger the browser's `popstate` event because
the Angular Router uses `pushState` and `replaceState`. This can be
confusing when calling `Location.go` and using `Location.subscribe`.
This commit clarifies the behavior of `Location.subscribe` and points
developers to the `onUrlChanges` subscription instead.

Fixes #12691

PR Close #42272

atscott committed:
docs: add note about redirects after absolute redirect (#42277)

This commit addresses the documentation portion of #19069.

PR Close #42277

atscott committed:
docs: document shallow equality emit for ActivatedRoute (#42280)

fixes #40769

PR Close #42280

atscott committed:
docs: add small clarity around navigationCancel (#42282)

Add small clarity to sentence in documentation for navigation cancel event
to indicate that router guards returning false or urlTree is only one of
several reasons a NavigationCancel event happens.

fixes #26613

PR Close #42282

atscott committed:
docs: fix documentation of guard execution order (#42288)

The current documentation of the `CanDeactivate`, `CanActivate`, and
`CanActivateChild` guard execution order is incorrect. This commit
corrects the documentation. For reference, see https://stackblitz.com/edit/angular-canactivatechild-nqp1f7

Fixes #28082

PR Close #42288

IgorMinar committed:
docs: clarify the definition of the bundle bundget defaults with strict mode (#42289)

Previously we didn't clarify that the stricter defaults have no impact on existing
projects, and we also didn't define the default values for bundle budgets.

Fixes #41966

PR Close #42289

Commit:
docs: add section on binding to passive events (#42292)

fixes #41622

PR Close #42292

zarend committed:
docs: update contributor instructions for installing java (#42294)

For angular contributors, updates the instructions for installing java
in the DEVELOPER.md. Adds more details about which version of java is
requried, and what uses java.

Resolves #29043

PR Close #42294

IgorMinar committed:
fix(docs-infra): increase the opacity of background-color for inline `code` blocks (#42297)

I intentionally did not change the font size as discussed in #41196
because the current
font size is already about the same as the normal text size.

Fixes #41196

PR Close #42297

Commit:
docs: remove references to CheckAlways and CheckOnce (#42300)

Fixes #29751

PR Close #42300

umairhm committed:
docs(core): improve applicationref.bootstrap docs

add proper examples and update usage notes
add references to the method at relevant places in the docs

petebacondarwin commented on packages/core/src/application_ref.ts:
```diff
@@ -682,20 +682,51 @@ export class ApplicationRef {
+   *
+   * ### Example
+   *
+   * Generally, we define the component to bootstrap in the bootstrap array of `NgModule`,
```
```suggestion
   * Generally, we define the component to bootstrap in the `bootstrap` array of `NgModule`,
```

petebacondarwin commented on packages/core/src/application_ref.ts:
````diff
@@ -682,20 +682,51 @@ export class ApplicationRef {
    *
-   * ### Example
-   * {@example core/ts/platform/platform.ts region='longform'}
+   * ```typescript
+   * ngDoBootstrap(appRef: ApplicationRef) {
+   *   // it can be a CSS selector
+   *   . . .
+   *   appRef.bootstrap(this.componentMap[componentName], '#root-element');
+   *   . . .
+
+   *   // or it can be a reference to a DOM node
+   *   . . .
+   *   const element = document.querySelector('#root-element');
+   *   appRef.bootstrap(this.componentMap[componentName], element);
+   *   . . .
+   * }
+   * ```
````
Ideally we need to put this into the actual example code so that it gets type-checked.
I propose that you add something like the following to the `core/ts/platform/platform.ts` example file:

```ts
export class AppModule2 extends AppModule {
  // #docregion elementSelector
  ngDoBootstrap(appRef: ApplicationRef) {
    this.fetchDataFromApi().then((componentName: string) => {
      appRef.bootstrap(this.componentMap[componentName], '#root-element');
    });
  }
  // #enddocregion elementSelector
}

export class AppModule3 extends AppModule {
  // #docregion elementInstance
  ngDoBootstrap(appRef: ApplicationRef) {
    this.fetchDataFromApi().then((componentName: string) => {
      const element = document.querySelector('#root-element');
      appRef.bootstrap(this.componentMap[componentName], '#root-element');
    });
  }
  // #enddocregion elementInstance
}
```

Then you can reference the two doc regions: `elementSelector` and `elementInstance`.

petebacondarwin reviewed (changes requested):
Thanks for recreating the PR @umairhm.

If you are happy with my suggestions, please create a fixup commit and push it to this branch. You can read more about how and why to do this here: https://github.com/angular/angular/blob/master/docs/FIXUP_COMMITS.md

alan-agius4 committed:
docs: update web-worker syntax (#42306)

With this change we update the worker initialization syntax to reflect that used in Angular CLI 12.

Closes #42302

PR Close #42306

atscott committed:
docs: clarify expectations for `ActivatedRoute` observables (#42316)

https://github.com/angular/angular/issues/16261#issuecomment-748289240

A couple important things to note about the behavior:

* The component is destroyed, meaning that all of its members will be destroyed as well, including any subscriptions to the route params
* This _does not_ mean that any `finalize` operators or `complete` on the subscription is called. This only happens if the source completes or you unsubscribe from the subscription. The documentation doesn't state that the `Router` will do this, though I can still understand why the behavior is confusing.

You can play around with these scenarios here:
https://stackblitz.com/edit/rxjs-hmpizx
* `complete` the source without unsubscribe: `next`, `complete` and `finalize` are all called, even when `complete` is called before the `next` from the `delay`
* `unsubscribe` from subscription without `complete` on the source: `finalize` happens, but the `complete` of the subscription does not, and neither does `next`

So even if the `Router` were to call `complete` on all of the `Observables` on an `ActivatedRoute`, you would see that any subscriptions with a `delay` operator would still get called.

resolves #16261

PR Close #42316

AndrewKushnir commented on packages/core/src/application_ref.ts:
```diff
@@ -682,20 +682,51 @@ export class ApplicationRef {
-   * Bootstrap a new component at the root level of the application.
+   * Bootstrap a component onto the element identified by its selector or, optionally, to a
+   specified
+   * element
```
```suggestion
   * element.
```

atscott committed:
docs: add link to the style precedence guide in ivy compatibility (#42317)

fixes #36508

PR Close #42317

josephperrott committed:
feat(dev-infra): add support for determining if builds should be stamped (#42319)

Add support for the build process to determine if the generated builds
should be stamped for release.

PR Close #42319

AndrewKushnir commented on packages/core/src/application_ref.ts:
```diff
@@ -682,20 +682,51 @@ export class ApplicationRef {
-   * When bootstrapping a new root component into an application, Angular mounts the
-   * specified application component onto DOM elements identified by the componentType's
-   * selector and kicks off automatic change detection to finish initializing the component.
+   * When bootstrapping an application with a component, Angular mounts the specified application
+   * component onto a DOM element, and kicks off automatic change detection to finish initializing
+   * the component.
+   *
+   * If not provided in the `rootSelectorOrNode` argument, the selector of the provided
+   * `componentOrFactory` will be used to match the DOM elements.
+   *
+   * Please note that if multiple elements match the selector, the first matched element will be
+   * used as a host element.
```
I'd propose to rephrase this a bit:
```suggestion
   * When bootstrapping a component, Angular mounts it onto a target DOM element and kicks off automatic
   * change detection. The target DOM element can be provided using the `rootSelectorOrNode` argument. If 
   * the target DOM element is not provided, Angular tries to find one on a page using a `selector` of the
   * component that is being bootstrapped (first matched element is used).
```

josephperrott committed:
feat(dev-infra): add `ng-dev build-and-link` command (#42319)

Add a command to build the release output without stamping for release
and link via `yarn link` the generated builds to a project provided.

PR Close #42319

AndrewKushnir commented on packages/core/src/application_ref.ts:
```diff
@@ -682,20 +682,51 @@ export class ApplicationRef {
+   * Generally, we define the component to bootstrap in the bootstrap array of `NgModule`,
+   * but it requires us to know the component while writing the application code.
+   *
+   * Imagine a situation where we have to wait for an API call to decide about the component to
+   * bootstrap. We can use the `ngDoBootstrap` hook of the `NgModule` and call this method to
```
```suggestion
   * Imagine a situation where we have to wait for an API call to decide which component to
   * bootstrap. We can use the `ngDoBootstrap` hook of the `NgModule` and call this method to
```

josephperrott committed:
build: properly handle whether builds should be stamped for release (#42319)

Properly build snapshot or release stamped versions of the builds based
on the provided context from the ng-dev toolset.

PR Close #42319

atscott committed:
docs: update date pipe api reference to indicate it is a pure pipe (#42320)

This note was included in #37099 but the PR was closed without being merged.

Fixes #33897

PR Close #42320

AndrewKushnir commented on packages/core/src/application_ref.ts:
```diff
@@ -682,20 +682,51 @@ export class ApplicationRef {
+   * ngDoBootstrap(appRef: ApplicationRef) {
+   *   // it can be a CSS selector
+   *   . . .
+   *   appRef.bootstrap(this.componentMap[componentName], '#root-element');
```
I think we should avoid using `this.componentMap` here if possible, so it's easier to understand what the code does. 
```suggestion
   *   appRef.bootstrap(ComponentOne, '#root-element');
```
and in the `ngDoBootstrap` code below you can have an if-else block:
```
ngDoBootstrap(appRef: ApplicationRef) {
  this.fetchDataFromApi().then((componentName: string) => {
    if (componentName === 'one') {
      // #docregion elementSelector
      appRef.bootstrap(ComponentOne, '#root-element');
      // #enddocregion elementSelector
    } else {
      // #docregion elementInstance
      const element = document.querySelector('#root-element');
      appRef.bootstrap(ComponentTwo, element);
      // #enddocregion elementInstance
    }
  });
}
```

umairhm replied:
@AndrewKushnir Thank you for the kind feedback.

For the `ngDoBootstrap` code, I think it will be better if we follow what @petebacondarwin mentioned above:

https://github.com/angular/angular/pull/42301#discussion_r638534546

petebacondarwin replied:
I agree that for the examples that are demonstrating the additional parameter, we can avoid the dynamic nature of the previous example:

```ts
export class AppModule2 {
  // #docregion elementSelector
  ngDoBootstrap(appRef: ApplicationRef) {
      appRef.bootstrap(AppComponent, '#root-element');
  }
  // #enddocregion elementSelector
}

export class AppModule3 {
  // #docregion elementInstance
  ngDoBootstrap(appRef: ApplicationRef) {
      const element = document.querySelector('#root-element');
      appRef.bootstrap(AppComponent, element);
    });
  }
  // #enddocregion elementInstance
}
```

AndrewKushnir reviewed (commented):
@umairhm thanks for creating this PR 👍

josephperrott committed:
docs: add explanation of how to use locally created build artifacts in a local project (#42319)

Add explanation of using `ng-dev build-and-link` to test local changes in another project.

Closes #39719

PR Close #42319

Commit:
docs: add example that exports a part of an animation for reuse (#42321)

Fixes #41334

PR Close #42321

josephperrott committed:
docs: update wording for angular libraries mapping peer dependencies (#42322)

Update the wording for Angular Libraries mapping of peer dependencies to make
it clearer that only the parts of @angular/* that the library depends on
should be marked as peer dependencies.

Closes #37304

PR Close #42322

atscott committed:
docs: add note about why module import order is important (#42323)

Fixes #30223

PR Close #42323

AndrewKushnir committed:
docs: remove `entryComponents` references from the custom elements guide (#42324)

Since `entryComponents` field is deprecated and no longer used in Ivy, this commit removes `entryComponents` references from the custom elements guide.

Closes #40341.

PR Close #42324

petebacondarwin committed:
docs: move "Browser Support" page in navigation tree (#42326)

Closes #39270

PR Close #42326

TeriGlover committed:
docs: Edit correction to remove jargon in Release section (#42029)

PR Close #42029

IgorMinar committed:
docs(common): rewrite docs for NgForOf#ngForTrackBy (#42329)

Clarify the prupose of the tracking function and document how to create a good one.

Fixes #40461

PR Close #42329

atscott committed:
docs: add docs for DebugElement (#42352)

fixes #36048

PR Close #42352

josephperrott committed:
docs: clarify the shortcomings of local variable technique for parent-child component interactions (#42331)

Closes #38302

PR Close #42331

IgorMinar committed:
docs: move the schematics under Angular Tools menu section (#42332)

In #39176 Alan noted that schematics is currently misplaced and should really be under the Angular Tools menu item.

In that issue, Alan also noted the "Angular Compiler" and "Angular CLI" are not refereced from here.
Since the issue was created "Angular Compiler" has already been moved under "Angular Tools".

Angular CLI is documented throughout the docs, and the "CLI Command Reference" under References is really just a reference
so it seems correctly located.

Fixes #39176

PR Close #42332

IgorMinar committed:
docs: move and update browser compatibility docs to browser-support guide (#42334)

I simplified the content significantly since most of it is not relevant any more.

Fixes #26525

PR Close #42334

swseverance committed:
docs: fix `BannerComponent` unit tests (#42336)

remove `async` and `await` from `BannerComponent` test because the
component uses an inline template and styles

create doc region in `banner-external.component.spec.ts` demonstrating
test setup that may fail due to a missing call to `.compileComponents()`
for a component with an external template and stylesheet

PR Close #42336

mgechev committed:
ci: update the github feature request process action and run on cron (#42337)

- Update to the latest hash
- Set the limit to 100 feature requests
- Run daily at 2pm

PR Close #42337

IgorMinar committed:
docs: add a note about <select> and nested <option>s in Ivy compat guide (#42338)

An internal change in Ivy has surfaced issues in previosly broken code. This change adds a note to the
Ivy compatibility guide as well as the TrackByFunction api docs.

Fixes #35896

PR Close #42338

AndrewKushnir committed:
docs: improve preloading component data chapter (#42340)

This commit updates the `Lazy-loading feature modules` guide (guide/lazy-loading-ngmodules) to fix example code that illustrates the process of preloading component data.

Closes #37113.

PR Close #42340

belgoros committed:
docs: edit project structure description (#42346)

- improve `app` folder description: an initial skeleton app project, also called angular-tour-of-heroes (in the src subfolder), there is no `angular-tour-of-heroes` in the `src` subfolder.
- an end-to-end test project (in the e2e subfolder) removed as does not exist/was not created
PR Close #42346

swseverance committed:
docs: add missing `await` to `BannerComponent` unit test (#42336)

PR Close #42336

swseverance committed:
docs: refactor `HeroDetailComponent` and unit test (#42349)

remove `@Input()` decorator from `hero` property because the component
is designed to get the hero via a service, not an input binding.

add `HTMLElement` type to `HeroDetailComponent` unit test

PR Close #42349

josephperrott committed:
fix(dev-infra): build with stamping for release builds (#42354)

`ng-dev release build` should always build with version stamping rather than
snapshot stamping.

PR Close #42354

atscott committed:
docs: clarify providedIn options (#42355)

The list for the possible options of providedIn was not totally clear. This commit ensures each possible value is included explicitly in the docs.

fixes #29330

PR Close #42355

zarend committed:
build: use wombat-dressing-room proxy for publishing releases (#42357)

In the scripts to release latest and next, use the npm proxy for publishing the
release.

PR Close #42357

zarend committed:
release: cut the v12.1.0-next.3 release (#42360)

atscott committed:
docs: clarify that `forRoot` can be called in other modules (#42361)

The faq on `forRoot` implies that you cannot and should not call `forRoot` for an import
outside the AppModule. This is technically incorrect. This commit clarifies
the statement and refers the reader to the hierarchical injectors guide.

fixes #29002

PR Close #42361

AndrewKushnir committed:
docs: clarify when the `ngDoBootstrap` hook is invoked (#42362)

Closes #35242.

PR Close #42362

wagnermaciel committed:
docs: explain what a deploy url is and how it differs from a base href (#42363)

fixes #34931

PR Close #42363

AndrewKushnir committed:
docs: use `json` pipe in Forms guide instead of a temporary diagnostic field (#42364)

Closes #37283.

PR Close #42364

zarend committed:
build: allow node v15 and above for aio (#42367)

Changes the engines rule in aio to accept any node version greater than
or equal to v14. This allows node v15 to work with aio.

addresses #42076

PR Close #42367

Commit:
docs: remove lightbox gif from Use Directives to Watch the DOM section. (#42369)
This gif is out of date and causing reader confusion.

Fixes #40193

PR Close #42369

IgorMinar committed:
docs: add a note about the ngc command and command line options (#42373)

It turns out that we don't document anywhere how ngc works and what options we support.
I added a brief section, without going too much into details because most users should not need to use ngc directly.

I also came across some bug in `ngc` and filed #42372.

Fixes #29623

PR Close #42373

atscott committed:
docs: add note to clarify that ngOnChanges is only called if inputs are bound (#42374)

fixes #23017

PR Close #42374

fusho-takahashi committed:
docs: fix a typo (#42384)

The 'B' in 'StackBlitz' is lowercased.
PR Close #42384

swseverance committed:
docs: update links to Testing Component Scenarios page (#42385)

PR Close #42385

gkalpak committed:
docs: fix typos in the `cli-builder` guide (#42386)

PR Close #42386

gkalpak committed:
build(docs-infra): remove unused `spy-directive.gif` image (#42388)

The only reference to the `spy-directive.gif` image was removed
in #42369, so the image is no longer used. This commit removes the file
from the repository.

PR Close #42388

atscott committed:
docs: cherry pick release notes for 12.0.2 to main (#42390)

PR Close #42390

atscott committed:
docs: change markdown table to HTML table in template typecheck doc (#42392)

PR Close #42392

atscott committed:
docs: clarify behavior of strict templates flags (#42392)

fixes #39355

PR Close #42392

IgorMinar committed:
docs: expand documentation for ROUTES (#42398)

Previously the docs were very minimalistic. The most important thing missing from the docs
was that people should primarily use higher level APIs instead of using ROUTES directly.

It would be nice to holistically overhaul more of the router API docs, but that's out of
scope of this change.

Fixes #39350

PR Close #42398

umairhm committed:
fixup! docs(core): improve applicationref.bootstrap docs
